client-v2/src/actions/Fronts.js

```diff
@@ -42,17 +53,34 @@ function fetchLastPressed(frontId: string): ThunkAction {
       });
 }
 
-function publishCollection(collectionId: string): ThunkAction {
+function publishCollection(collectionId: string, frontId: string): ThunkAction {
   return (dispatch: Dispatch) =>
     publishCollectionApi(collectionId)
-      .then(() =>
+      .then(() => {
         dispatch(
           batchActions([
             publishCollectionSuccess(collectionId),
             recordUnpublishedChanges(collectionId, false)
           ])
-        )
-      )
+        );
+
+        return new Promise(resolve => setTimeout(resolve, 10000))
```

Comment:
Not sure if this is the best way to do this, suggestions are welcomed. I copied the value of the timeout from the old app.

Reply:
Could we use the constant declared further down for this value?

Reply:
It's sort of a pain that promises aren't cancelable. It looks like this code will just create a promise every time we publish a collection. That is potentially quite infrequently so it should be fine but part of me things we should move towards using something like `lodash.throttle` here, although that would be harder to implement. Maybe that's overkill but if we end up doing something like publishing on edit in some circumstance in future then this may be a memory leak. Be interested to have @jonathonherbert 's eye on this. Happy to let this go as long as we've thought about it and don't mind.

Reply:
Having talked to editorial on multiple desks I think it's going to be unlikely that we'll have to ever publish anything on edit so maybe this will be fine, what does @jonathonherbert think?

Reply:
Then I'm happy with this if @jonathonherbert is 👍